tsdb/head_append.go

```diff
@@ -793,17 +793,19 @@ func (a *headAppender) AppendHistogramCTZeroSample(ref storage.SeriesRef, lset l
 		// We set it to true to make this implementation as close as possible to the float implementation.
 		isOOO, _, err := s.appendableHistogram(ct, zeroHistogram, a.headMaxt, a.minValidTime, a.oooTimeWindow)
 		if err != nil {
-			s.Unlock()
 			if errors.Is(err, storage.ErrOutOfOrderSample) {
+				s.Unlock()
 				return 0, storage.ErrOutOfOrderCT
 			}
 		}
```

Comment:
Maybe we should `return err` instead? I think that's what's missing. Otherwise the check failed but we proceed.

Same applies to the next block.